dGame/Entity.cpp

```diff
@@ -946,12 +925,17 @@ void Entity::WriteBaseReplicaData(RakNet::BitStream* outBitStream, eReplicaPacke
 			outBitStream->Write0(); //No ldf data
 		}
 
-		if (m_Trigger != nullptr && m_Trigger->events.size() > 0) {
-			outBitStream->Write1();
-		} else {
+		TriggerComponent* triggerComponent;
+		if (TryGetComponent(COMPONENT_TYPE_TRIGGER, triggerComponent)) {
+			// has trigger component, check to see if we have events to handle
+			auto* trigger = triggerComponent->GetTrigger();
```

Comment:
Feels like putting what's in this scope into a Serialize method in the trigger component would make sense. Yet the trigger component is weird as it always has at least one bit associated with it regardless, so idk.

Reply:
There's technically some other part of the trigger component that MAY need to be serialized via `WriteComponents`, but that needs more investigation into the client. I did have this in `TriggerComponent->WriteBaseReplicaData(bitStream)` and it was about same amount of ugly, but less obvious as to what was going on, so I chose to make it happen right here as the top level